lib/apiservers/service/restapi/handlers/common.go

```diff
@@ -186,18 +187,34 @@ func getVCHConfig(op trace.Operation, d *data.Data, validator *validate.Validato
 	return vchConfig, nil
 }
 
-func getAddresses(vchConfig *config.VirtualContainerHostConfigSpec) (dockerHost, adminPortal string) {
-	if client := vchConfig.ExecutorConfig.Networks["client"]; client != nil {
-		if publicIP := client.Assigned.IP; publicIP != nil {
-			var dockerPort = opts.DefaultTLSHTTPPort
-			if vchConfig.HostCertificate.IsNil() {
-				dockerPort = opts.DefaultHTTPPort
-			}
+func getAddresses(executor *management.Dispatcher, vchConfig *config.VirtualContainerHostConfigSpec) (string, string, error) {
+	clientNet := vchConfig.ExecutorConfig.Networks["client"]
+	if clientNet != nil {
+		clientIP := clientNet.Assigned.IP
+
+		if ip.IsUnspecifiedIP(clientIP) {
+			return "", "", fmt.Errorf("No client IP address assigned")
+		}
 
-			dockerHost = fmt.Sprintf("%s:%d", publicIP, dockerPort)
-			adminPortal = fmt.Sprintf("https://%s:%d", publicIP, constants.VchAdminPortalPort)
+		hostIP := clientIP.String()
+		// try looking up preferred name irrespective of CAs, if available
+		// once we found a preferred address, we use that instead of the assigned client IP address
+		if cert, err := vchConfig.HostCertificate.X509Certificate(); err == nil {
+			hostIP = executor.GetTLSFriendlyHostIP(clientIP, cert, vchConfig.CertificateAuthorities)
 		}
+
+		var dockerPort string
```

Comment:
Why not just define `dockerPort` as an integer? That should avoid the need to `fmt.Sprintf("%d",` below.

Reply:
it's consistent with CLI implementation:
```
	if !conf.HostCertificate.IsNil() {
		d.DockerPort = fmt.Sprintf("%d", opts.DefaultTLSHTTPPort)
	} else {
		d.DockerPort = fmt.Sprintf("%d", opts.DefaultHTTPPort)
	}
```
CLI is setting the `d.DockerPort` field of type string in executor, and in API we're not using the docker IP/port from executor fields. 
Fixing it now.